### PR TITLE
Added the mbs-frontend deployment.

### LINKED ISSUE #46: mbs_frontend CRD deployment config
As part of the mbs_frontend CRD we should define a deployment config.

## Acceptance criteria

* Add a deployment config for mbs_frontend
* Add task to deploy mbs_frontend
* Add local tests for deployment
* Add cluster tests for deployment

## Definition of Done

* Acceptance criteria are met
* Local tests have been run and pass - if failed it is re-opened/fixed and passed

### LINKED ISSUE #49: mbs_frontend CRD service deployment
As part of the mbs_frontend CRD we should define a service deployment config.

## Acceptance criteria

* Add a ingress config for mbs_frontend
* Add a deployment config for mbs_frontend
* Add task to deploy ingress config
* Add local tests
* Add cluster tests

## Definition of Done

* Acceptance criteria are met
* Local tests have been run and pass - if failed it is re-opened/fixed and passed

### PR DESCRIPTION
**Pull request by AdamSaleh**
Fixes #49

## Verification Steps

Tests should pass.

## Additional Notes

Not sure if fixes #46 or if that was part of #90

### CODE REVIEW

File: mbox-operator/roles/mbs-frontend/tasks/main.yml [hunk 1]
```diff
@@ -48,3 +48,19 @@
         src: "{{ mbs_frontend_dir.path }}/mbs-configmap.yaml"
 
 - include_tasks: cert.yml
+
+- block:
+    - name: ensure mbs-backend deployment
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-frontend/tasks/main.yml:53]:
```suggestion
    - name: ensure mbs-frontend deployment
```

File: mbox-operator/roles/mbs-frontend/tasks/main.yml [hunk 2]
```diff
@@ -48,3 +48,19 @@
...
+      template:
+        src: deployment.yaml.j2
+        dest: "{{ mbs_dir.path }}/deployment.yaml"
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-frontend/tasks/main.yml:56]: This should fix the error
```suggestion
        dest: "{{ mbs_frontend_dir.path }}/deployment.yaml"
```

File: mbox-operator/roles/mbs-frontend/tasks/main.yml [hunk 3]
```diff
@@ -48,3 +48,19 @@
...
+    - k8s:
+        src: "{{ mbs_dir.path }}/deployment.yaml"
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-frontend/tasks/main.yml:58]: And another one here
```suggestion
        src: "{{ mbs_frontend_dir.path }}/deployment.yaml"
```

File: mbox-operator/roles/mbs-frontend/tasks/main.yml [hunk 4]
```diff
@@ -48,3 +48,19 @@
...
+        state: present
+        namespace: "{{ meta.namespace }}"
+        wait: true
+
+- name: delete temporary component directory
+  file:
+    path: "{{ mbs_dir.path }}"
```
Comment (Zlopez) [inline mbox-operator/roles/mbs-frontend/tasks/main.yml:65]: And here
```suggestion
    path: "{{ mbs_frontend_dir.path }}"
```

**Comment by AdamSaleh** [comment]
Need help. Can't figure out what is causing the deployment to fail. Seeing this error:
```
{   "changed":true,
   "duration":120,
   "method":"patch",
   "msg":"Resource update timed out",
   "result":{      "apiVersion":"apps/v1",
      "kind":"Deployment",
      "metadata":{         "annotations":{
            "deployment.kubernetes.io/revision":"1"
         
},
         "creationTimestamp":"2020-06-11T08:12:16Z",
         "generation":1,
         "labels":{
            "app":"mbs-frontend",
            "service":"mbs-frontend"
         
},
         "name":"mbs-frontend",
         "namespace":"osdk-test",
         "ownerReferences":[            {
               "apiVersion":"apps.fedoraproject.org/v1alpha1",
               "kind":"MBMbsFrontend",
               "name":"mb-mbs-frontend",
               "uid":"e6766e94-f102-401e-87cd-5c917e0f5268"
            
}
         
],
         "resourceVersion":"2229",
         "selfLink":"/apis/apps/v1/namespaces/osdk-test/deployments/mbs-frontend",
         "uid":"b979c471-4795-4e7e-9f75-98580ee919c3"
      
},
      "spec":{         "progressDeadlineSeconds":600,
         "replicas":1,
         "revisionHistoryLimit":10,
         "selector":{            "matchLabels":{
               "app":"mbs-frontend"
            
}
         
},
         "strategy":{            "rollingUpdate":{
               "maxSurge":"25%",
               "maxUnavailable":"25%"
            
},
            "type":"RollingUpdate"
         
},
         "template":{            "metadata":{               "creationTimestamp":null,
               "labels":{
                  "app":"mbs-frontend"
               
}
            
},
            "spec":{               "containers":[                  {                     "image":"quay.io/fedora/mbs-frontend:latest",
                     "imagePullPolicy":"Always",
                     "name":"mbs-frontend",
                     "ports":[                        {
                           "containerPort":8443,
                           "protocol":"TCP"
                        
}
                     
],
                     "resources":{

                     
},
                     "terminationMessagePath":"/dev/termination-log",
                     "terminationMessagePolicy":"File",
                     "volumeMounts":[                        {
                           "mountPath":"/etc/module-build-service",
                           "name":"config-volume",
                           "readOnly":true
                        
},
                        {
                           "mountPath":"/etc/mbs-frontend",
                           "name":"frontend-config-volume",
                           "readOnly":true
                        
},
                        {
                           "mountPath":"/etc/cacert",
                           "name":"cacert-volume",
                           "readOnly":true
                        
},
                        {
                           "mountPath":"/etc/servicecert",
                           "name":"service-cert-volume",
                           "readOnly":true
                        
}
                     
]
                  
}
               
],
               "dnsPolicy":"ClusterFirst",
               "restartPolicy":"Always",
               "schedulerName":"default-scheduler",
               "securityContext":{

               
},
               "terminationGracePeriodSeconds":30,
               "volumes":[                  {                     "configMap":{
                        "defaultMode":420,
                        "name":"mbs-configmap"
                     
},
                     "name":"config-volume"
                  
},
                  {                     "configMap":{
                        "defaultMode":420,
                        "name":"mbs-frontend-configmap"
                     
},
                     "name":"frontend-config-volume"
                  
},
                  {                     "name":"cacert-volume",
                     "secret":{
                        "defaultMode":420,
                        "secretName":"mbs-frontend-client-cert"
                     
}
                  
},
                  {                     "name":"service-cert-volume",
                     "secret":{
                        "defaultMode":420,
                        "secretName":"mbs-frontend-service-cert"
                     
}
                  
}
               
]
            
}
         
}
      
},
      "status":{         "conditions":[            {
               "lastTransitionTime":"2020-06-11T08:12:16Z",
               "lastUpdateTime":"2020-06-11T08:14:05Z",
               "message":"ReplicaSet \"mbs-frontend-57794b4f8b\" has successfully progressed.",
               "reason":"NewReplicaSetAvailable",
               "status":"True",
               "type":"Progressing"
            
},
            {
               "lastTransitionTime":"2020-06-11T08:16:23Z",
               "lastUpdateTime":"2020-06-11T08:16:23Z",
               "message":"Deployment does not have minimum availability.",
               "reason":"MinimumReplicasUnavailable",
               "status":"False",
               "type":"Available"
            
}
         
],
         "observedGeneration":1,
         "replicas":1,
         "unavailableReplicas":1,
         "updatedReplicas":1
      
}
   
}
}
```

**Pull request merged**